tests/rptest/tests/admin_uuid_operations_test.py

```diff
@@ -82,18 +89,211 @@ def _uuids_updated():
+        self.admin.decommission_broker(ghost_node_id, node=to_stop)
+
+        self.logger.debug(f"...and wait for the cluster to become healthy")
+
+        self.redpanda.wait_until(self.redpanda.healthy,
+                                 timeout_sec=30,
+                                 backoff_sec=1)
+
+        self.logger.debug(
+            "Check that all this state sticks across a rolling restart")
+
+        self.redpanda.rolling_restart_nodes(self.redpanda.nodes,
+                                            auto_assign_node_id=True)
+
+        after_restart = self.admin.get_broker_uuid(to_stop)
+
+        assert after_restart[
+            'node_id'] == initial_to_stop_id, f"Bad node id after override: {after_restart['node_id']}"
+        assert after_restart[
+            'node_uuid'] == old_uuid, f"Bad node uuid after override: {after_restart['node_uuid']}"
+
+        self.redpanda.wait_until(self.redpanda.healthy,
+                                 timeout_sec=30,
+                                 backoff_sec=1)
+
+    @cluster(num_nodes=3)
+    def test_force_uuid_override_multinode(self):
+        to_stop = self.redpanda.nodes[1:]
+        initial_to_stop_ids = [self.redpanda.node_id(n) for n in to_stop]
+
+        self.logger.debug("Kill one node, all is good")
+
+        self._restart_node(to_stop[0], drop_disk=True)
+
+        uuids = wait_until_result(
+            lambda: self._uuids_updated(),
+            timeout_sec=30,
+            backoff_sec=2,
+            err_msg="Node was unable to join the cluster")
+
+        ghost_node_id = self.admin.get_broker_uuid(to_stop[0])['node_id']
+
+        old_uuids = {}
+        for n in uuids:
+            id = n['node_id']
+            if id in initial_to_stop_ids:
+                old_uuids[id] = n['uuid']
+
+        assert len(
+            old_uuids) == 2, f"Unexpected old_uuids: {json.dumps(old_uuids)}"
+
+        self.logger.debug("Drop another node, this time restart should fail")
+
+        for n in to_stop:
+            with expect_exception(TimeoutError, lambda _: True):
+                self._restart_node(n, drop_disk=True)
+
+        current_uuids = [self.scrape_uuid(n) for n in to_stop]
+        assert len(current_uuids
+                   ) == 2, f"Missing some UUIDs: {json.dumps(current_uuids)}"
+
+        self.logger.debug(
+            "Restart both nodes again, with overrides. Keep both disks")
+
+        self.redpanda.restart_nodes(
+            to_stop,
+            override_cfg_params=dict(node_id_overrides=[
+                dict(current_uuid=current_uuids[n],
+                     new_uuid=old_uuids[initial_to_stop_ids[n]],
+                     new_id=initial_to_stop_ids[n])
+                for n in range(0, len(to_stop))
+            ], ))
+
+        self.logger.debug(f"Decommission ghost node [{ghost_node_id}]...")
+
+        self.admin.decommission_broker(ghost_node_id, node=to_stop[0])
```

Comment:
Do we need to wait for controller log leadership before issuing this? Or maybe issue with `wait_until` succeeds.

Reply:
>Do we need to wait for controller log leadership...?

I believe so. Moving after the health check and waiting is probably good.

Reply:
done